server/analysis/gpr/gpr_models.py

```diff
@@ -110,19 +107,14 @@ def _get_kernel_kwargs(self, **kwargs):
             {
                 'input_dim': X_dim,
                 'ARD': False
-            },
-            {
-                'input_dim': X_dim,
-            },
+            }
         ]
 
     def _build_kernel(self, kernel_kwargs, **kwargs):
-        k0 = gpflow.kernels.Exponential(**kernel_kwargs[0])
-        k1 = gpflow.kernels.White(**kernel_kwargs[1])
+        k = gpflow.kernels.Matern12(lengthscales=2, **kernel_kwargs[0])
         if kwargs.pop('optimize_hyperparameters'):
-            k0.lengthscales.transform = gpflow.transforms.Logistic(
+            k.lengthscales.transform = gpflow.transforms.Logistic(
                 *self._LENGTHSCALE_BOUNDS)
-        k = k0 + k1
```

Comment:
Removing the white kernel is a little risky for tuning the real DBMS.

Reply:
I remove the white kernel to make it consistent with old GPR model. According to the experiment, the white kernel has little effect on the results. Also, it will have one more hyper-parameter to tune for white kernel.  For now I just remove it and make it simple. We can add it back if we find it helps when tuning real databases.

Reply:
(1) I understand that you're changing the kernel from Exponential to Matern12 to make the GPFlow implementation equivalent to gp_tf.py, but how does the performance of the two different kernels compare? If the Exponential kernel performs better than the Matern12 kernel then we should change the kernel in gp_tf.py to Exponential.

(2) Instead of completely removing the white noise kernel from the codebase can you move it into a new subclass of BasicGP?

Reply:
Thanks for the comments.
(1) Matern12 are almost same as Exponential kernel in GPFlow. The only difference is length_scale. Matern12 is k(r) = σ² exp{-r} and Exponential is k(r) = σ² exp{-0.5 * r}, where r is |x-x'|/length_scale. This is why I set default length_scale of Matern12 to 2, and set it to 1 for Exponential. Actually, from other materials, Exponential kernel is same as Matern12: https://www.mathworks.com/help/stats/kernel-covariance-function-options.html

(2) It sounds good. I will add a new subclass.